app/assets/javascripts/services/miq_service.js

```diff
@@ -158,4 +158,20 @@ ManageIQ.angular.app.service('miqService', ['$timeout', '$document', '$q', 'API'
       miqService.sparkleOff();
     }
   };
+
+  this.redirectBack = function(message, flashType, redirectUrl, reload) {
+    var flash = { message: message };
+
+    flash.level = flashType;
+
+    miqFlashLater(flash);
+
+    $timeout(function() {
+      $window.location.href = redirectUrl;
+
+      if(reload) {
```

Comment:
Please refer de8063c

Reload is essential for reloading the treeview -- a nice bug reported by @h-kataria and resolved with a fairly simple change.